components/config/useLocalPreferences.tsx

```diff
@@ -0,0 +1,90 @@
+import {
+  buildGraphicPreferenceKeyForLabel,
+  labelsRequiringMediaFilter,
+  GraphicMediaFilter,
+} from '@/common/labels'
+import { useLocalStorage } from 'react-use'
+
+export type GraphicMediaFilterPreference = Record<GraphicMediaFilter, boolean>
+
+// This is the container hook for interfacing on top of localstorage and storing device level preferences
+// There may be various types of preferences that can be stored here and ideally each preference category
+// will have its own hook that will utilize this
+export const useLocalPreferences = () => {
+  const initialValue: Record<string, Record<string, boolean>> = {
+    graphicMediaPrefs: {},
+  }
+
+  labelsRequiringMediaFilter.forEach((label) => {
+    ;(['blur', 'grayscale', 'translucent'] as GraphicMediaFilter[]).forEach(
+      (filter) => {
+        initialValue.graphicMediaPrefs[
+          buildGraphicPreferenceKeyForLabel(label, filter)
+        ] = true
+      },
+    )
+  })
```

Comment:
I find this a bit difficult to read!  Would be sweet if we could get it to be something like:
```ts
labelsRequiringMediaFilter.forEach((label) => {
  GRAPHIC_MEDIA_FILTERS.forEach((filter) => {
    const key = buildGraphicPreferenceKeyForLabel(label, filter)
    initialValue.graphicMediaPrefs[key] = true
  })
})
```